.github/workflows/ci.yaml

```diff
@@ -36,7 +36,7 @@ jobs:
         sudo mv /etc/nix/nix.conf{,.bak}
         nix \
           --extra-experimental-features 'nix-command flakes' \
-          build .#checks.x86_64-darwin.integration
+          build .#checks.aarch64-darwin.integration
```

Comment:
I used x86_64 because aarch64 images didn't exist at the time, I agree this needs to be updated. What would you think about using something dynamic here, like this?

That way if a GHA runner happens to switch architectures again, it will default to intel but use aarch64 when appropriate. (Not sure, but I believe aarch64 can run either, which is why I've defaulted to x86 here. Also, I find the output of `arch -x86_64 /bin/bash -c arch` confusing).

```
ARCH=x86_64
if [ "$(arch)" = arm64 ]; then
    ARCH=aarch64
fi
nix build ... .#checks."${ARCH}"-darwin.integration
```